spec/rspec/support/object_inspector_spec.rb

```diff
@@ -67,6 +67,30 @@ def with_date_loaded
         end
       end
 
+      context 'with Hash Objects' do
+        let(:hash) { { 'key' => BigDecimal('3.3') } }
+        let(:formatted_hash) { ObjectInspector.inspect(hash) }
+
+        it 'recursively uses itself to inspect objects within it' do
+          in_sub_process_if_possible do
+            require 'bigdecimal'
+            expect(formatted_hash).to include('=> 3.3 (#<BigDecimal')
+          end
+        end
+      end
+
+      context 'with Array Objects' do
+        let(:array) { [[BigDecimal('3.3')]] }
+        let(:formatted_array) { ObjectInspector.inspect(array) }
+
+        it 'recursively uses itself to inspect objects within it' do
+          in_sub_process_if_possible do
+            require 'bigdecimal'
+            expect(formatted_array).to include('[[3.3 (#<BigDecimal')
+          end
+        end
+      end
```

Comment:
Both of these specs show only part of the output.  It's important that the full structure of the array/hash is formatted properly so I think I'd like to see `expect(formatted_hash|array).to eq(expected_output)` rather than just part of it.

Also, I think it would work better to use `Time` objects -- the output is more deterministic than with `bigdecimal` and doesn't require the special `in_sub_process_if_possible` handling.